pandas/tests/series/test_api.py

```diff
@@ -117,9 +118,78 @@ def test_to_sparse_pass_name(self):
         result = self.ts.to_sparse()
         assert result.name == self.ts.name
 
+    def test_constructor_dict(self):
+        d = {'a': 0., 'b': 1., 'c': 2.}
+        result = self.Series(d)
+        expected = self.Series(d, index=sorted(d.keys()))
+        tm.assert_series_equal(result, expected)
+
+        result = self.Series(d, index=['b', 'c', 'd', 'a'])
+        expected = self.Series([1, 2, np.nan, 0], index=['b', 'c', 'd', 'a'])
+        tm.assert_series_equal(result, expected)
```

Comment:
This doesn't check sparseness. I might modify assert_series_equal to dispatch to assert_sp_series_equal if *both* are SparseSeries.